torch/lib/nccl/Makefile

```diff
@@ -51,6 +51,12 @@ NVCUFLAGS  := -ccbin $(CXX) $(NVCC_GENCODE) -lineinfo -std=c++11 -maxrregcount 9
 # Use addprefix so that we can specify more than one path
 LDFLAGS    := $(addprefix -L,${CUDA_LIB}) -lcudart -lrt
 
+# If CUDA < 8.0, add workaround C++ flags
+CUDA_MAJOR_LT_8 := $(shell [ $(CUDA_MAJOR) -lt 8 ] && echo true)
+ifeq ($(CUDA_MAJOR_LT_8), true)
+CXXFLAGS += -D_FORCE_INLINES -D_MWAITXINTRIN_H_INCLUDED -D__STRICT_ANSI__
+endif
```

Comment:
This file shouldn't be changed. It's part of the NCCL repo that we subtree

Reply:
Mmm... I see lots of changes in your subtree w.r.t. https://github.com/NVIDIA/nccl master

Reply:
Hmm I thought it's still frozen, @soumith is that no longer the case?

Reply:
@apaszke we did make quite a few changes to the Makefile around CUDA detection and arch selection

Reply:
@jpuigcerver Why is `_MWAITXINTRIN_H_INCLUDED` and `__STRICT_ANSI__` needed?

Reply:
@apaszke I got this recipe a long time ago from Tensorflow, and I've been used it since then.
https://github.com/tensorflow/tensorflow/issues/1066

Reply:
@jpuigcerver well this issue isn't very convincing. It looks like a hack that has to be applied sometimes, and wasn't included into TF by default. The flags you want to add are not documented online and are internal to compiler headers, so I would rather avoid patching them. It might be a bug in the default Ubuntu compiler, consider upgrading it to a newer version.

Reply:
Apparently NCCL guys are not in favor if fixing this either: https://github.com/NVIDIA/nccl/issues/75

They suggest to make a nvcc alias that defines the required flags.

Reply:
There you go. I'd rather avoid globally hacking all compilers that might not really need those defines.

Reply:
i think it's fine. this is for CUDA 7.5 support, i.e. the hack applies only when we compile under that setting. And I know (and remember from cutorch days) that there aren't any visible side-effects